@here/olp-sdk-core/lib/utils/TileKey.ts

```diff
@@ -0,0 +1,421 @@
+     *
+     * @param quadkey The quadkey to convert.
+     * @returns A new instance of `TileKey`.
+     */
+    static fromQuadKeyString(key: string): TileKey {
+        if (key === "-") {
+            return TileKey.fromRowColumnLevel(0, 0, 0);
+        }
+
+        const level = key.length;
+        let row = 0;
+        let column = 0;
+        // tslint:disable:no-bitwise
+        for (let i = 0; i < key.length; ++i) {
+            const mask = 1 << i;
+            const d = parseInt(key.charAt(level - i - 1), 10);
+            if (d & 0x1) {
+                column |= mask;
+            }
+            if (d & 0x2) {
+                row |= mask;
+            }
+        }
+        // tslint:enable:no-bitwise
+        return TileKey.fromRowColumnLevel(row, column, level);
+    }
+
+    /**
+     * Creates a tile key from a numeric Morton code representation.
+     *
+     * You can convert a tile key into a numeric Morton code with [[mortonCode]].
+     *
+     * @param key The Morton code to be converted.
+     * @returns A new instance of [[TileKey]].
+     */
+    static fromMortonCode(key: number): TileKey {
+        let level = 0;
+        let row = 0;
+        let column = 0;
+        let quadKey = key;
+        // tslint:disable:no-bitwise
+        while (quadKey > 1) {
+            const mask: number = 1 << level;
+
+            if (quadKey & 0x1) {
+                column |= mask;
+            }
+            if (quadKey & 0x2) {
+                row |= mask;
+            }
+
+            level++;
+            // tslint:disable-next-line: no-magic-numbers
+            quadKey = (quadKey - (quadKey & 0x3)) / SUBTILES_COUNT;
+        }
+        // tslint:enable:no-bitwise
+        const result = TileKey.fromRowColumnLevel(row, column, level);
+        result.m_mortonCode = key;
+        return result;
+    }
+
+    /**
+     * Returns the number of available columns at a given level.
+     *
+     * This is 2 to the power of the level.
+     *
+     * @param level The level for which to return the number of columns.
+     * @returns The available columns at the given level.
+     */
+    static columnsCount(level: number): number {
+        return Math.pow(2, level);
+    }
+
+    /**
+     * Returns the number of available rows at a given level.
+     *
+     * This is 2 to the power of the level.
+     *
+     * @param level The level for which to return the number of rows.
+     * @returns The available rows at the given level.
+     */
+    static rowsCount(level: number): number {
+        return Math.pow(2, level);
+    }
+
+    /**
+     * Computes the Morton code of the parent tile key of the given Morton code.
+     *
+     * Note: The parent key of the root key is the root key itself.
+     *
+     * @param mortonCode A Morton code, for example, obtained from [[mortonCode]].
+     * @returns The Morton code of the parent tile.
+     */
+    static parentMortonCode(mortonCode: number): number {
```

Comment:
Is this method useful at all or should you rather return the parent as TileKey?

Reply:
yes, it will be better. updated, thanks.